src/models/Rule/Rule.spec.ts

```diff
@@ -0,0 +1,101 @@
+import Rule from './Rule';
+import Symbolizer from '../Symbolizer/Symbolizer';
+import Filter from '../Filter/Filter';
+import ScaleDenominator from '../ScaleDenominator/ScaleDenominator';
+
+it('is defined', () => {
+  expect(Rule).toBeDefined();
+});
+
+describe('Constructor', () => {
+  it('can be created with only without filter or scaledenominator', () => {
+    const symbolizer = new Symbolizer();
+    const rule = new Rule(symbolizer);
+    expect(rule).toBeDefined();
+    expect(rule.symbolizer).toBe(symbolizer);
+  });
+
+  it('can be created with style and filter', () => {
+    const symbolizer = new Symbolizer();
+    const filter = new Filter();
+    const rule = new Rule(symbolizer, filter);
+    expect(rule).toBeDefined();
+    expect(rule.symbolizer).toBe(symbolizer);
+    expect(rule.filter).toBe(filter);
+  });
+
+  it('can be created with style and scaleDenominator', () => {
+    const symbolizer = new Symbolizer();
+    const scaleDenominator = new ScaleDenominator();
+    const rule = new Rule(symbolizer, undefined, scaleDenominator);
+    expect(rule).toBeDefined();
+    expect(rule.symbolizer).toBe(symbolizer);
+    expect(rule.scaleDenominator).toBe(scaleDenominator);
+  });
+
+  it('can be created with style, filter and scaleDenominator', () => {
+    const symbolizer = new Symbolizer();
+    const filter = new Filter();
+    const scaleDenominator = new ScaleDenominator();
+    const rule = new Rule(symbolizer, filter, scaleDenominator);
+    expect(rule).toBeDefined();
+    expect(rule.symbolizer).toBe(symbolizer);
+    expect(rule.filter).toBe(filter);
+    expect(rule.scaleDenominator).toBe(scaleDenominator);
+  });
+});
+
+describe('Accessors', () => {
```

Comment:
Good that this is tested. Any chance of making this a less manual work? both when creating classes and testing them?

Reply:
1. Didn't found anything for my editor.
2. We should discuss if these properties need to be private anyway. If we use `getters` and `setters` they should do something. Otherwise we can just leave these properties public.
 I really don't want that JAVAish overhead but i was playing around with TypeScript features. :wink:

Reply:
Not related to the generation of test files, but this [extension](https://marketplace.visualstudio.com/items?itemName=DSKWRK.vscode-generate-getter-setter) looks promising for the work with typescript in combination with visual studio code.